code/game/objects/items/devices/chameleonproj.dm

```diff
@@ -123,4 +123,8 @@
 				else
 					spawn(25) can_move = 1
 			step(src,direction)
-		return
\ No newline at end of file
+		return
+
+/obj/effect/dummy/chameleon/Destroy()
+	master.disrupt(0)
+	return ..()
```

Comment:
This whole file is in bad shape. Calling disrupt will del the dummy effect before its Destroy call finishes, making it return null. The projector item itself doesn't have a Destroy, so there will be a circular reference between it and its dummy effect if it gets qdel'd. Also, a bunch of dels instead of qdels.